Mathlib/GroupTheory/GroupExtension/Abelian.lean

```diff
@@ -0,0 +1,114 @@
+/-
+Copyright (c) 2024 Yudai Yamazaki. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yudai Yamazaki
+-/
+
+import Mathlib.GroupTheory.GroupExtension.Basic
+import Mathlib.RepresentationTheory.GroupCohomology.LowDegree
+
+/-!
+# Lemmas about extensions by Abelian groups
+
+This file gives lemmas about group extensions `1 → N → E → G → 1` that hold when `N` is Abelian.
+
+For the main definitions, see `Mathlib/GroupTheory/GroupExtension/Defs.lean`. For basic lemmas about
+group extensions in general, see `Mathlib/GroupTheory/GroupExtension/Basic.lean`.
+
+## Main definition
+
+- `SemidirectProduct.conjClassesEquivH1 (φ : G →* MulAut N)`: the bijection between the
+  `N`-conjugacy classes of splittings associated to the semidirect product `G ⋊[φ] N` and
+  $H^1 (G, N)$
+
+## References
+
+* [Kenneth S. Brown, *Cohomology of groups*][brown1982]
+
+-/
+
+namespace SemidirectProduct
+
+variable {N G : Type} [CommGroup N] [Group G] (φ : G →* MulAut N)
+
+/-- The `ℤ`-linear `G`-representation associated to a multiplicative action -/
+noncomputable def toRep : Rep ℤ G :=
+  @Rep.ofMulDistribMulAction G N _ _ (MulDistribMulAction.compHom N φ)
+
+theorem toRep_ρ_apply_apply (g : G) (n : Additive N) :
+    ((toRep φ).ρ g) n = Additive.ofMul (φ g (Additive.toMul n)) := rfl
+
+/-- Returns the 1-cocycle corresponding to a splitting. -/
+def splittingToOneCocycle (s : (toGroupExtension φ).Splitting) :
+    groupCohomology.oneCocycles (toRep φ) where
+  val g := Additive.ofMul (α := N) (s g).left
+  property := by
+    rw [groupCohomology.mem_oneCocycles_iff]
+    intro g₁ g₂
+    rw [toRep_ρ_apply_apply, toMul_ofMul, map_mul, mul_left, mul_comm, ← ofMul_mul, right_splitting]
+
+/-- Returns the splitting corresponding to a 1-cocycle. -/
+def splittingOfOneCocycle (f : groupCohomology.oneCocycles (toRep φ)) :
```

Comment:
Similarly this could be `Splitting.ofOneCocycle` for the anonymous dot notation.